plugin/pkg/scheduler/algorithm/priorities/priorities.go

```diff
@@ -170,6 +170,77 @@ func (n *NodeLabelPrioritizer) CalculateNodeLabelPriority(pod *api.Pod, machines
 	return result, nil
 }
 
+// This is a reasonable size range of all container images. 90%ile of images on dockerhub drops into this range.
+const minImgSize int64 = 23000000   // min 23MB
+const maxImgSize int64 = 1000000000 // max 1000MB
+
+// ImageLocalityPriority is a priority function that favors nodes with requested containers' images of a Pod.
+// It will detect whether the requested images are present on a node, and then calculate a score
+// range from 0 to 10 based on the sum size of those images:
+// - If all the images are not present on a node, this node will be given the lowest priority.
+// - If some of the images are present on a node, the larger their sizes' sum, the higher the node's priority.
+func ImageLocalityPriority(pod *api.Pod, machinesToPods map[string][]*api.Pod, podLister algorithm.PodLister, nodeLister algorithm.NodeLister) (schedulerapi.HostPriorityList, error) {
+	var score int
+	var sumSizeMap = make(map[string]int64)
+
+	nodes, err := nodeLister.List()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, container := range pod.Spec.Containers {
+		for _, node := range nodes.Items {
+
+			// Check if this container's image is present and get its size.
+			imageSize := checkContainerImageOnNode(node, container)
+			// Add this size to the total result of this node.
+			sumSizeMap[node.Name] += imageSize
+		}
+	}
+
+	result := []schedulerapi.HostPriority{}
+	// score int - scale of 0-10
+	// 0 being the lowest priority and 10 being the highest,
+	for nodeName, sumSize := range sumSizeMap {
+		switch {
+		case sumSize == 0 || sumSize < minImgSize:
+			// score == 0 means non of the images required by this Pod is present on this node,
```

Comment:
"score == 0 means none of the images required by this pod are present on this node or the total size of the images present is too small to be taken into further consideration."